src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/UserController.kt

```diff
@@ -78,16 +79,15 @@ class UserController(
         val userName = userName(context)
         val roleReadingScopes = roleReadingScopes(context)
 
-        val user = userRepository.getUserByUsername(userName)
+        val internalUser = userRepository.getUserByUsername(userName)
         if (roleReadingScopes.any())
         {
-            val allRoles = userRepository.getRolesForUser(userName)
-            val roles = filteredRoles(allRoles, roleReadingScopes)
-            return user.copy(roles = roles)
+            val roles = filteredRoles(internalUser.roles, roleReadingScopes)
+            return internalUser.copy(roles=roles).toUser()
```

Comment:
Why do you here copy and then convert to `User`, whereas in the else branch you convert to `User` and then set the roles?

Reply:
Because the InternalUser can't have a null `roles` property, but see if you find this change clearer